### ISSUE TITLE
Sass Interpolation - no-missing-end-of-source-newline (returns error)

### ISSUE THREAD

**Issue by Ne-Ne** [stylelint/stylelint#2037]
> Describe the issue. Is it a bug or a feature request (new rule, new option, etc.)?

Returns error when running following code below.

> Which rule, if any, is this issue related to?

no-missing-end-of-source-newline

> What CSS is needed to reproduce this issue?

```scss
@mixin breakpoint($value) {

    $str: breakpoint($value);

    // If $str is still an empty string, no media query is needed
    @if $str == "" {
        @content;
    }

        // Otherwise, wrap the content in a media query
    @else {
        @media screen and #{$str} {
            @content;
        }
    }
}
```

> What stylelint configuration is needed to reproduce this issue?

```json
{
  "rules": {
    "no-missing-end-of-source-newline": true,
  }
}
```

> Which version of stylelint are you using?

`7.5.0`

> How are you running stylelint: CLI, PostCSS plugin, Node API?

`gulp-stylelint 3.4.0`

> Does your issue relate to non-standard syntax (e.g. SCSS, nesting, etc.)?

Sass Interpolation

> What did you expect to happen?

No warnings to be flagged.

> What actually happened (e.g. what warnings or errors you are getting)?

```shell
Unexpected missing end-of-source newline   **no-missing-end-of-source-newline**
```

**Comment by jeddy3** [comment]
@Ne-Ne It's impossible to tell from your CSS snippet if you _actually have_ a newline at the end of your source (file). Can you please confirm your source (file) has a trailing newline. If you do have one, can you create a test repo please to help us resolve this issue?

**Comment by jeddy3** [comment]
Closing as not reproducible locally. @Ne-Ne Please feel free to reopen if you can create a reproducible repo.
